### ISSUE TITLE
New color picker improvements, second round

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#2699]
Follow up to #2679 

See also Slack discussion https://wordpress.slack.com/archives/C02QB2JS7/p1504633514000516

I'd like to propose to consider a few further improvements for the color picker:
- add some indication about the currently selected color _before_ the set of buttons: "visually it’s clear, but if you are on a screen reader you may have to go through a number of items before you get to the one selected."
- ideally, colors should have a human-understandable named color. While teh required format is hex or rgb or maybe other that will be added, I guess many users (not just a11y users) would just like to pick up a color. Other applications already refer to colors with a name:

![screen_shot_2017-09-05_at_20 03 37](https://user-images.githubusercontent.com/1682452/30168383-8a8543aa-93e9-11e7-887e-eac047e5eaed.png)

also for color variations:

<img width="557" alt="screen_shot_2017-09-05_at_20 10 40" src="https://user-images.githubusercontent.com/1682452/30168384-8a89e5b8-93e9-11e7-9945-bd38aad78057.png">

thi could be implemented requiring to specify a color name in `gutenberg_color_palette()`.

**Comment by gschoppe** [comment]
This change would be useful to agencies, to provide branding hints... e.g. "Primary Brand Color" "Secondary Brand Color", "Green Accent", etc

**Comment by afercia** [comment]
Yep, not to mention accessibility because a string like, for example, #8ed1fc gets read out as:
`number eight ed one f c` which is not so useful for assistive technology users.

![screen shot 2017-09-09 at 12 14 29](https://user-images.githubusercontent.com/1682452/30239257-bf6661e2-9558-11e7-9d77-d74ea9aba21e.png)

**Comment by gschoppe** [comment]
Sorry, yes, I meant that as "in addition to a11y concerns, it would assist agencies"

**Comment by afercia** [comment]
Sure 🙂  And your point makes perfectly sense. I haven't thought at that and seems to me one more good reason to consider named colors.

**Comment by gschoppe** [comment]
Just a side note, thank you for the incredible a11y work you've been doing for the team.  I work for an agency that often contracts out with an accessibility-oriented firm, and the level of thought that you put into your work is very heartening to see. I'm well versed in how little fun it is to boot up a screen reader and test interfaces over and over again that were designed by sighted developers for mouse-driven interactions.

**Comment by afercia** [comment]
@gschoppe thanks. I share that feeling on a daily basis... 😬

**Comment by gschoppe** [comment]
The label could expand, tool-tip-like on focus/hover (although that often leaves touch users in the cold). and, of course, it needn't be visible for screen readers. However, human-readable, visible color labels would also assist sighted users with any form of color impairment.

**Comment by afercia** [comment]
Gutenberg tooltips work in a way that they use the same text used for the `aria-label` attribute, so that would work for sighted users and screen reader users. Worth also considering not all screen reader users are completely blind; many users with low vision use screen readers as an aid.

**Comment by gschoppe** [comment]
how do gutenberg tooltips perform on touch devices, like phones?

**Comment by afercia** [comment]
@gschoppe I think that's untested and still to address.

**Comment by afercia** [comment]
Re: better identifying colors, apart from giving them a human-understandable named color, here's some feedback from users about giving them also a tooltip, which makes sense to me, see https://make.wordpress.org/accessibility/2017/11/28/screen-reader-user-experience-with-gutenberg/

> the user can select colours for text and background. It would be useful to have the name visible on hover. This will be useful for colourblind people.

**Comment by afercia** [comment]
Reported by one of the accessibility testers:

- What the user hears is the hexadecimal code of the color. It will be better if the name of the color was used instead of the code hex.

<img width="919" alt="screen shot 2018-03-20 at 00 55 36" src="https://user-images.githubusercontent.com/1682452/37628364-7ad65a24-2bd9-11e8-8f2b-2d2d25680e94.png">

**Comment by afercia** [comment]
Thanks everyone for the discussion, feedback, and thanks @jorgefilipecosta very good job!
